core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -55,6 +63,8 @@
   private boolean sendHeadersCalled;
   private boolean closeCalled;
   private Compressor compressor;
+  private boolean messageSent;
+  private boolean internalClosed;
```

Comment:
The `ServerStream` API states that close() will close the stream for reading and writing, but did not specify what happens if we keep interacting with it after calling close(). `InProcessTransport` and `AbstractStream`  happen to both ignore all interactions after `close`. I didn't want to rely on what seems like behavior, and opted to never touch the stream again after calling `close()`. If you're ok with adding that behavior to the API, this flag can be removed.